### PR TITLE
Implemented setting value via updater function

### PR DESCRIPTION
**Pull request by AlexShukel**

### CODE REVIEW

File: src/hooks/useStock.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { MutableRefObject, useCallback, useRef } from 'react';
+import { MutableRefObject, SetStateAction, useCallback, useRef } from 'react';
```
Comment (ArtiomTr) [inline src/hooks/useStock.ts:1]: Possibly incorrect import from react

File: src/hooks/useStockState.ts [hunk 2]
```diff
@@ -4,7 +4,8 @@ import { StockContext } from '../components/StockContext';
 import { Stock } from './useStock';
 import { useStockValue } from './useStockValue';
 
-export type SetAction<V> = (value: V) => void;
+export type SetStateAction<S> = S | ((value: S) => S);
+type Dispatch<A> = (value: A) => void;
```
Comment (ArtiomTr) [inline src/hooks/useStockState.ts:8]: These lines should be inside `types` folder, not here

**Review by ArtiomTr — CHANGES_REQUESTED** [review]
Some mistakes

**Comment by ArtiomTr** [comment]
@AlexShukel also there is no tests for `useStock.ts` updater

File: src/hooks/useStockState.ts (hunk 2 above)
Comment (AlexShukel) [reply src/hooks/useStockState.ts]: Should I move Dispatch type in single file?

**Review by ArtiomTr — APPROVED** [review]
Ok

**Pull request merged**